src/data/datasets/acdc_adapt_dataset.py

```diff
@@ -0,0 +1,64 @@
+import torch
+import nibabel as nib
+import numpy as np
+import pandas as pd
+from pathlib import Path
+
+from src.data.datasets import BaseDataset
+from src.data.transforms import Compose, ToTensor
+
+
+class AcdcAdaptDataset(BaseDataset):
+    """The dataset of the Automated Cardiac Diagnosis Challenge (ACDC) in MICCAI 2017
+    for the self-supervised learning.
+
+    Ref: 
+        https://www.creatis.insa-lyon.fr/Challenge/acdc/index.html
+
+    Args:
+        data_split_file_path (str): The data split file path.
+        preprocess (BoxList): The preprocessing techniques applied to the data.
+        transforms (BoxList): The self-supervised transforms applied to the data.
+    """
+
+    def __init__(self, data_split_file_path, preprocess, transforms, **kwargs):
+        super().__init__(**kwargs)
+        if self.type == 'train':
+            data_split_file = pd.read_csv(data_split_file_path)
+            patient_dirs = map(
+                Path,
+                data_split_file[
+                    (data_split_file.type == 'train') | (data_split_file.type == 'valid')
+                ].path
+            )
+            self.data_paths = tuple(
+                data_path
+                for patient_dir in patient_dirs
+                for data_path in sorted(patient_dir.glob('**/*frame??.nii.gz'))
```

Comment:
Can we use each frame in the 4D MRI data instead of only ED and ES two frames when we pre-train/adapt the network with ACDC dataset?

Reply:
I think it's a good point in practice; however, we want to claim/verify that it's our adapting strategy works, not because of the more data we use. So I think we should only use ED and ES frame.

Reply:
Got it.